app/plugin.go

```diff
@@ -394,3 +398,48 @@ func (a *App) GetPlugins() (*model.PluginsResponse, *model.AppError) {
 
 	return resp, nil
 }
+
+// notifyPluginEnabled notifies web clients for all cluster peers if
+// manifest version is same across all peers, including self.
+func (a *App) notifyPluginEnabled(manifest *model.Manifest) *model.AppError {
+	if manifest.HasClient() {
+		var statuses model.PluginStatuses
+		expectedPeers := 1
+		actualPeers := 0
+
+		if a.Cluster != nil {
+			var err *model.AppError
+			statuses, err = a.Cluster.GetPluginStatuses()
+			if err != nil {
+				return err
+			}
+			if infos := len(a.Cluster.GetClusterInfos()); infos > 0 {
+				expectedPeers = infos
+			}
+		}
+
+		localStatus, err := a.GetPluginStatus(manifest.Id)
+		if err != nil {
+			return err
+		}
+		statuses = append(statuses, localStatus)
+
+		for _, status := range statuses {
+			// Count consistent peers
+			isPeerConsistent := status.Version == manifest.Version &&
+				localStatus.State == model.PluginStateRunning && localStatus.State == status.State
+			if status.PluginId == manifest.Id && isPeerConsistent {
+				actualPeers++
+			}
+		}
+
+		// Notify all cluster peer clients.
+		if actualPeers == expectedPeers {
```

Comment:
Instead of counting, could we flip things around to instead just return if we find a peer whose manifest doesn't match my own? This naturally collapses when there are no peers, and should simplify things tremendously.

Reply:
+1

Reply:
funny enough, i had this flipped initially, but switched to this to guard against installing a plugin and then quickly enabling it, before the other peer has a chance to install it. In which case, it will only find the plugin for this peer and end up broadcasting to all clients, which may or may not result in a bundle 404.

However, even with this guard in place, I found hitting the `enable` endpoint returning a 500 because the peer i ended up calling `enable` on didn't have the plugin installed (yet), so don't think this is a full working solution. 

Since this race condition will most likely occur in a case where one is programmatically installing a plugin and then enabling it, i'm inclined to revert back to the original solution i had and document the case it's still failing to guard against.